src/parse_instruction.py

```diff
@@ -313,8 +322,6 @@ def expect(n: str) -> str:
             if "$" in reg:
                 # If there is a second $ in the word, it's a symbol
                 value = AsmGlobalSymbol(word)
-            elif reg in arch.aliased_regs:
-                value = arch.aliased_regs[reg]
             else:
                 value = Register(reg)
```

Comment:
maybe detect_bare_reg here, to make the logic less error prone? I agree it shouldn't be needed, though

Reply:
I'm not 100% sure what you mean here:

- This part of the code needs to be able to handle symbolic register names from asm patterns like `$y`, so `value = Register(reg)` cannot be replaced with `value = replace_bare_reg(AsmGlobalSymbol(reg), arch)`, because `replace_bare_reg` won't turn `y` into `Register("y")`. 

- I don't think `value = AsmGlobalSymbol(word)` here needs a `replace_bare_reg`: the replacement is done when `value` is *used* not when it is constructed. (see the `# Global symbol` section)

- I could handle `Register` input values in `replace_bare_reg`, and then change this code to do `value = replace_bare_reg(Register(reg), arch)` -- but I don't think that is more or less clean than what I've done on my latest commit?

Reply:
The last point is what I was thinking, under the misapprehension that `replace_bare_reg` already accepted Register inputs. The way you have it with the latest commit is good, though!